### PR TITLE
TCOMP-326: Adds JavaScript client tests

### PR DESCRIPTION
**Pull request by talend-glorieux**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

No JavaScript client tests.

**What is the new behavior?**

JavaScript client tests

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:

* Adds yarn.lock
* Adds test stack librairies
* Fixes license field
* Adds basic coverage support

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/472/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/472/)

File: services/components-rest-client-js/yarn.lock [hunk 1]
```diff
@@ -1,3 +1,442 @@
 # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (sgandon) [inline services/components-rest-client-js/yarn.lock]: If this autogenerated, could it be possible to not store it on git and generate it on the fly as required ?

**Comment by sgandon** [comment]
how are these test launched and what do they require to be launched ?

**Comment by talend-glorieux** [comment]
You'll need a nodejs stack to launch the tests. I'm working on running the tests on the CI using a nodejs docker image. 
With that stack in place, a simple `yarn run test` will run the test suite and `yarn run cov` will give you the coverage report.

**Comment by sgandon** [comment]
Can you check with dataprep, I am pretty sure they have such an image for their tests.

**Comment by sgandon** [comment]
Can you get one of the front-end guys to review this PR? I am not qualified to do so.

**Comment by talend-glorieux** [comment]
Can I get some @Talend/frontend people to review this ?

File: services/components-rest-client-js/spec/tcomp_spec.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+				done();
+			});
+		});
+	});
+
+	describe('ValidateProperties', function () {
+		checkArgs('ValidateProperties');
+
+		it('should validate a property', function (done) {
+			fetchMock.post('/api/properties/test/validate', {});
```
Comment (frassinier) [inline services/components-rest-client-js/spec/tcomp_spec.js]: Maybe we could externalize theses URLs into variables?

**Comment by sgandon** [comment]
@talend-glorieux feel free to merge it if you are satisfied with the review.

**Comment by talend-glorieux** [comment]
@sgandon I don't have the rights to merge it. Could you do it?

**Pull request merged**